target/linux/generic/hack-6.12/736-net-phy-realtek-add-rtl8251l-rtl8254b-rtl8261be-rtl8261n.patch

```diff
@@ -0,0 +1,704 @@
+Date: Sat, 18 Oct 2025 23:15:58 +0200
+Subject: [PATCH] phy-realtek: add rtl8251l, rtl8254b, rtl8261be, rtl8261n, 
+ rtl8264 and rtl8264b
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+This change adds/moves support for rtl8251l, rtl8254b, rtl8261be, rtl8261n,
+rtl8264 and rtl8264b phys into common realtek phy driver.
+
+Signed-off-by: Balázs Triszka <info@balika011.hu>
+
+---
+ drivers/net/phy/realtek/Makefile       |   2 +
+ drivers/net/phy/realtek/realtek.h      |  10 +
+ drivers/net/phy/realtek/realtek_main.c | 607 +++++++++++++++++++++++++
+ 3 files changed, 619 insertions(+)
+
+--- a/drivers/net/phy/realtek/Makefile
++++ b/drivers/net/phy/realtek/Makefile
+@@ -1,4 +1,6 @@
+ # SPDX-License-Identifier: GPL-2.0
+ realtek-y			+= realtek_main.o
++realtek-y			+= phy_patch.o
++realtek-y			+= phy_patch_rtl826x.o
+ realtek-$(CONFIG_REALTEK_PHY_HWMON) += realtek_hwmon.o
+ obj-$(CONFIG_REALTEK_PHY)	+= realtek.o
+--- a/drivers/net/phy/realtek/realtek.h
++++ b/drivers/net/phy/realtek/realtek.h
+@@ -5,6 +5,16 @@
+ 
+ #include <linux/phy.h>
+ 
++#include "phy_patch.h"
++#include "phy_patch_rtl826x.h"
++
++struct rtl826x_priv {
++	rt_phy_patch_db_t *patch;
++	bool pnswap_tx : 1;
++	bool pnswap_rx : 1;
++	bool patch_rtk_serdes : 1;
++};
++
+ int rtl822x_hwmon_init(struct phy_device *phydev);
+ 
+ #endif /* REALTEK_H */
+--- a/drivers/net/phy/realtek/realtek_main.c
++++ b/drivers/net/phy/realtek/realtek_main.c
+@@ -15,6 +15,7 @@
+ #include <linux/delay.h>
+ #include <linux/clk.h>
+ #include <linux/string_choices.h>
++#include <linux/crc32.h>
+ 
+ #include "realtek.h"
+ 
+@@ -158,6 +159,7 @@
+ 
+ #define PHYID_RTL_GENERIC			0x001cc800
+ #define PHYID_RTL8201CP				0x00008201
++#define PHYID_RTL8264B				0x001cc813
+ #define PHYID_RTL8201F				0x001cc816
+ #define PHYID_RTL8226				0x001cc838
+ #define PHYID_RTL8221B				0x001cc840
+@@ -179,11 +181,17 @@
+ #define PHYID_RTL8366S				0x001cc960
+ #define PHYID_RTL8366RB				0x001cc961
+ #define PHYID_RTL8224				0x001ccad0
++#define PHYID_RTL8264				0x001ccaf2
++#define PHYID_RTL8261N				0x001ccaf3
+ #define PHYID_RTL9000A				0x001ccb00
+ 
+ /* RTL8211E and RTL8211F support up to three LEDs */
+ #define RTL8211x_LED_COUNT			3
+ 
++#define RTL8261N_SERDES_GLOBAL_CFG		0xc1
++#define   RTL8261N_SERDES_GLOBAL_CFG_HSO_INV			BIT(7)
++#define   RTL8261N_SERDES_GLOBAL_CFG_HSI_INV			BIT(6)
++
+ MODULE_DESCRIPTION("Realtek PHY driver");
+ MODULE_AUTHOR("Johnson Leung");
+ MODULE_LICENSE("GPL");
+@@ -1525,6 +1533,115 @@ static int rtl8251b_c45_match_phy_device
+ 	return rtlgen_is_c45_match(phydev, PHYID_RTL8251B, true);
+ }
+ 
++static int rtl8251l_match_phy_device(struct phy_device *phydev,
++				     const struct phy_driver *phydrv)
++{
++	if (!rtlgen_is_c45_match(phydev, PHYID_RTL8261N, true))
++		return 0;
++
++	if (!phydev->mdio.bus->read_c45)
++		return 0;
++
++	int data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x103);
```

Comment:
Maybe we should printk this (somewhere, not necessarily here) to help refine the matching logic / debug #20294?